UIforETW/ChildProcess.cpp

```diff
@@ -112,10 +114,11 @@ DWORD ChildProcess::ListenerThread()
 	return 0;
 }
 
-
+_Pre_satisfies_( hProcess_ == 0 )
+_Success_( return )
 bool ChildProcess::Run(bool showCommand, std::wstring args)
 {
-	assert(!hProcess_);
+	ATLASSERT(!hProcess_);

```

Comment:
- `assert( )` calls abort (via _wassert), and breaks in the middle of the CRT, which means that you have to find the spot where the assertion actually fired (small deal, but annoying). It also means that hitting `Ignore` doesn't actually let the program continue.
- `ATLASSERT( )`, which is defined as `#define ATLASSERT(expr) _ASSERTE(expr)`, calls `_CrtDbgReport( )` and then `_CrtDbgBreak( )`, which allows the program to continue, and also breaks right into your code.
- `MFC`'s `ASSERT( )` is in between the two.


Kinda related, but interesting reading about ASSERT:
- [Press Cancel to cancel (or Cancelable Asserts)](https://lanzkron.wordpress.com/2013/10/02/press-cancel-to-cancel-or-cancelable-asserts/)